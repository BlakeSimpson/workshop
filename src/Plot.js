/**
 * This component should render a plot with Plotly, taking these three props:
 *
 * - xData: the data to be rendered on the x-axis as an array
 * - yData: The data to be rendered on the y-axis as an array
 * - type:  The type of plot we want Plotly to render
 *
 * To see an example of how you should use Plotly look at the index.html file in the public/ folder.
 * (and feel free to delete the code in there)
 */
import React, { Component } from 'react';
import Plotly from 'plotly.js/dist/plotly.js';
