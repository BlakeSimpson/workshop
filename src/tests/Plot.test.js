import React from 'react';
import { shallow, mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Plot from '../Plot';

// Mock the Plotly module (it doesn't like being snapshotted)
jest.mock('plotly.js/dist/plotly.js')
const Plotly = require('plotly.js/dist/plotly.js')

// Before each test, reset the mock
beforeEach(() => {
	Plotly.newPlot = jest.fn()
})

// Fixtures
const xData = [1, 2, 3, 4, 5];
const yData = [1, 4, 9, 16, 25];

it('should render a div', () => {
	const tree = toJson(shallow(<Plot xData={[1, 2, 3, 4, 5]} yData={[1, 4, 9, 16, 25]} type="scatter" />));
	expect(tree).toMatchSnapshot();
});

it('should render a plot with Plotly', () => {
	mount(<Plot xData={xData} yData={yData} type="scatter" />);
	expect(Plotly.newPlot.mock.calls[0][1]).toEqual([{
		x: xData,
		y: yData,
		type: "scatter"
	}]);
});

it('should re-render the Plot when the data changes', () => {
	const newXData = [10, 11, 12];
	const newYData = [100, 121, 144];
	const comp = mount(<Plot xData={xData} yData={yData} type="scatter" />);
	comp.setProps({
		xData: newXData,
		yData: newYData,
	})
	expect(Plotly.newPlot).toHaveBeenCalledTimes(2);
	expect(Plotly.newPlot.mock.calls[1][1]).toEqual([{
		x: newXData,
		y: newYData,
		type: "scatter"
	}]);
})

it('should not re-render the Plot if the passed data is the same', () => {
	const comp = mount(<Plot xData={xData} yData={yData} type="scatter" />);
	comp.setProps({
		xData,
		yData,
	})
	expect(Plotly.newPlot).toHaveBeenCalledTimes(1);
})
