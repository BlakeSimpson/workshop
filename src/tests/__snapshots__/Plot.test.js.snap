exports[`test should render a div 1`] = `<div />`;
